lib/active_admin/views/index_as_table.rb

```diff
@@ -283,26 +283,6 @@ def selectable_column
           end
         end
 
-        def index_column(start_value = 1)
-          column "#", class: "col-index", sortable: false do |resource|
-            @collection.offset_value + @collection.index(resource) + start_value
-          end
-        end
-
-        # Display a column for the id
-        def id_column
-          raise "#{resource_class.name} has no primary_key!" unless resource_class.primary_key
-          column(resource_class.human_attribute_name(resource_class.primary_key), sortable: resource_class.primary_key) do |resource|
-            if controller.action_methods.include?("show")
-              link_to resource.id, resource_path(resource), class: "resource_id_link"
-            elsif controller.action_methods.include?("edit")
-              link_to resource.id, edit_resource_path(resource), class: "resource_id_link"
-            else
-              resource.id
-            end
```

Comment:
I see that you changed all the above code with `auto_link resource, resource.id`. But the `auto_link` helper does a bit more stuff like checking authorization. Is this a separate bug fix?

Reply:
I could not just move the existing code. For example, if I create a table of posts on the User page, the links for Post id: 123 would be to `/user/123`. So in order to generate the right links, I would have to get the resource config using. This logic is already implemented in `auto_link`, so I chose to just use that. The additional authorisation checks seem like a free bonus.

I took me a while to recall this conclusion, so I had to do some mental backtracking, and I reverted the change in the process. Now I have re-implemented the old solution with the necessary changes but without using auto_link. Let me know, what you prefer.

Reply:
Arg! I cannot figure out how to get the `active_admin_config` instance inside `ActiveAdmin::Views::TableFor`. Can I ask the audience? 😅

Reply:
The new version looks good to me!